src/meta/src/manager/relation.rs

```diff
@@ -0,0 +1,68 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use risingwave_pb::catalog::{Sink, Table};
+
+// This enum is used in order to re-use code in `DdlServiceImpl` for creating MaterializedView and
+// Sink.
+pub enum Relation {
```

Comment:
cc @st1page, does this design LGTY?

Reply:
No, I think here the code re-using is for creating the streaming job on the Compute Node. So the "Relation" here seems just a stream destination job.

Reply:
@yezizp2012 I did consider adding `Source`, but for another reason. `Relation` can also be used to reduce code duplication in [CatalogManager](https://github.com/singularity-data/risingwave/blob/main/src/meta/src/manager/catalog.rs). `fn start_create_[source/sink/table]_procedure()` and `fn cancel_create_[source/sink/table]_procedure()` since these methods are practically the same. 

@st1page for me to be sure, are you ok with this design?